File: School source codes/Lab 5/Processgrade.cs
using System;
using System.Linq;

class MainClass {
  public static void Main (string[] args) {
    int grade1,grade2,grade3,grade4,grademax,grademin;
    double gradeaverage;
    Console.WriteLine ("Enter your grades: ");
    grade1 = Convert.ToInt32(Console.ReadLine());
    grade2 = Convert.ToInt32(Console.ReadLine());
    grade3 = Convert.ToInt32(Console.ReadLine());
    grade4 = Convert.ToInt32(Console.ReadLine());
    int[] gradearray = {grade1,grade2,grade3,grade4};
    grademin = gradearray.Min();
    grademax = gradearray.Max();
    gradeaverage = (grade1 + grade2 + grade3 + grade4)/4;
    Console.WriteLine("You entered: " +grade1 + ", " + grade2 + ", " + grade3 + ", " + grade4);
    Console.WriteLine("Highest grade: " + grademax);
    Console.WriteLine("Lowest grade: " + grademin);
    Console.WriteLine("Average grade: " + gradeaverage);
  }
}

File: School source codes/Lab 5/Processgrade.rb
puts "Enter your grades: "
grade1 = gets.chomp.to_i
grade2 = gets.chomp.to_i
grade3 = gets.chomp.to_i
grade4 = gets.chomp.to_i
gradearray = Array[grade1,grade2,grade3,grade4]
grademin = gradearray.min
grademax = gradearray.max
gradeaverage = (grade1 + grade2 + grade3 + grade4) / 4;
puts "You entered: #{grade1}, #{grade2}, #{grade3}, #{grade4}"
puts "Highest grade: #{grademax}"
puts "Lowest grade: #{grademin}"
puts "Average grade: #{gradeaverage}"